sdk/spring/azure-spring-cloud-stream-binder-eventhubs/src/main/java/com/azure/spring/eventhub/stream/binder/EventHubMessageChannelBinder.java

```diff
@@ -56,8 +56,12 @@ protected MessageHandler createProducerMessageHandler(ProducerDestination destin
         handler.setSendTimeout(producerProperties.getExtension().getSendTimeout());
         handler.setSendFailureChannel(errorChannel);
         if (producerProperties.isPartitioned()) {
-            handler.setPartitionKeyExpressionString(
+            if (null != producerProperties.getPartitionKeyExpression()) {
```

Comment:
personally I would not like this expression, put this *null* first, better to use more common/straight way to check